include/util/struct_packing.hpp

```diff
@@ -0,0 +1,19 @@
+#ifndef STRUCT_PACKING_HPP
+#define STRUCT_PACKING_HPP
+
+/*
+    Macro to pack struct/class in both environment
+    Usage:  PACK( 
+        struct MyStruct 
+        { 
+            int Property1;
+        });
+*/
+
+#ifdef _MSC_VER
+#define PACK( input ) __pragma( pack(push, 1) ) input __pragma( pack(pop) )
```

Comment:
The problem is not a packing by itself and also warnings are not very helpful. The real problem is in non-aligned access like in http://coliru.stacked-crooked.com/a/9fa8abe86e51c5ca
that will be a SIGBUS on some non-Intel platforms.

Such alignment problems can be detected  in run-time with ubsan, and there is no big difference between  silent packing with pragmas and ignored warnings.